cache/in-memory/src/event/guild.rs

```diff
@@ -0,0 +1,400 @@
+            application_id: guild.application_id,
+            banner: guild.banner,
+            default_message_notifications: guild.default_message_notifications,
+            description: guild.description,
+            discovery_splash: guild.discovery_splash,
+            explicit_content_filter: guild.explicit_content_filter,
+            features: guild.features,
+            icon: guild.icon,
+            joined_at: guild.joined_at,
+            large: guild.large,
+            max_members: guild.max_members,
+            max_presences: guild.max_presences,
+            member_count: guild.member_count,
+            mfa_level: guild.mfa_level,
+            name: guild.name,
+            nsfw_level: guild.nsfw_level,
+            owner: guild.owner,
+            owner_id: guild.owner_id,
+            permissions: guild.permissions,
+            preferred_locale: guild.preferred_locale,
+            premium_subscription_count: guild.premium_subscription_count,
+            premium_tier: guild.premium_tier,
+            rules_channel_id: guild.rules_channel_id,
+            splash: guild.splash,
+            system_channel_id: guild.system_channel_id,
+            system_channel_flags: guild.system_channel_flags,
+            unavailable: guild.unavailable,
+            verification_level: guild.verification_level,
+            vanity_url_code: guild.vanity_url_code,
+            widget_channel_id: guild.widget_channel_id,
+            widget_enabled: guild.widget_enabled,
+        };
+
+        self.0.unavailable_guilds.remove(&guild.id);
+        self.0.guilds.insert(guild.id, guild);
+    }
+}
+
+impl UpdateCache for GuildCreate {
+    fn update(&self, cache: &InMemoryCache) {
+        if !cache.wants(ResourceType::GUILD) {
+            return;
+        }
+
+        cache.cache_guild(self.0.clone());
+    }
+}
+
+impl UpdateCache for GuildDelete {
+    fn update(&self, cache: &InMemoryCache) {
+        fn remove_ids<T: Eq + Hash, U>(
+            guild_map: &DashMap<GuildId, HashSet<T>>,
+            container: &DashMap<T, U>,
+            guild_id: GuildId,
+        ) {
+            if let Some((_, ids)) = guild_map.remove(&guild_id) {
+                for id in ids {
+                    container.remove(&id);
+                }
+            }
+        }
+
+        if !cache.wants(ResourceType::GUILD) {
+            return;
+        }
+
+        let id = self.id;
+
+        cache.0.guilds.remove(&id);
+
+        if cache.wants(ResourceType::CHANNEL) {
+            remove_ids(&cache.0.guild_channels, &cache.0.channels_guild, id);
+        }
+
+        if cache.wants(ResourceType::EMOJI) {
+            remove_ids(&cache.0.guild_emojis, &cache.0.emojis, id);
+        }
+
+        if cache.wants(ResourceType::ROLE) {
+            remove_ids(&cache.0.guild_roles, &cache.0.roles, id);
+        }
+
+        if cache.wants(ResourceType::VOICE_STATE) {
+            // Clear out a guilds voice states when a guild leaves
```

Comment:
This comment I wrote months ago doesn’t seem needed given everything happening around it.